ambry-clustermap/src/main/java/com/github/ambry/clustermap/RecoveryTestClusterManager.java

```diff
@@ -52,7 +53,19 @@ public class RecoveryTestClusterManager implements ClusterMap {
   public PartitionId getPartitionIdFromStream(InputStream stream) throws IOException {
     DuplicatingInputStream duplicatingInputStream = new DuplicatingInputStream(stream);
     duplicatingInputStream.mark(0);
-    PartitionId partitionIdStatic = staticClusterManager.getPartitionIdFromStream(duplicatingInputStream);
+    PartitionId partitionIdStatic = null;
+    try {
+      partitionIdStatic = staticClusterManager.getPartitionIdFromStream(duplicatingInputStream);
+    } catch (IOException e) {
+      // for composite blob, it has the list of Blob IDs. It may not exist in the static cluster map.
+      // swallow the exception and try with helixClusterManager
+    }
+
+    if (partitionIdStatic == null) {
+      duplicatingInputStream.reset();
+      partitionIdStatic = helixClusterManager.getPartitionIdFromStream(duplicatingInputStream);
```

Comment:
This code could be moved in the catch block of the try-catch above. 
And are we sure that the helixClusterManager will always have the partitiionID. Even if it does, can it throw some excpetion that must be caught ?

Reply:
If we remove the code to the catch, it has the dependency on the implementation of the staticClusterManager.getPartitionIdFromStream.
If the staticClusterManager.getPartitionIdFromStream decides to "don't throw exception" but just return "null", we'll miss the case.

Check "if ==null" is more generic. 

If helixClusterManager returns null or exception, caller will fail somewhere.

Reply:
cool thanks